### PR TITLE
Task server not node

### PR DESCRIPTION
**Pull request by mjeronimo**
## Description of contribution in a few bullet points

* This PR addresses issue #44. The TaskServer is no longer a ROS Node, but accepts a rclcpp::Node::SharedPtr reference to use. 
* While making this change, I also updated the TaskClient and Robot classes to also accept an rclcpp:Node::SharedPtr reference for consistency.

### CODE REVIEW

File: nav2_mission_executor/include/nav2_mission_executor/mission_executor.hpp [hunk 1]
```diff
@@ -15,21 +15,24 @@
...
 namespace nav2_mission_executor
 {
 
-class MissionExecutor : public nav2_tasks::ExecuteMissionTaskServer
+class MissionExecutor : public rclcpp::Node
 {
 public:
   MissionExecutor();
 
   nav2_tasks::TaskStatus execute(
```
Comment (SteveMacenski) [inline nav2_mission_executor/include/nav2_mission_executor/mission_executor.hpp]: If this is no longer an implementation of a task, I think the `Execute()` method should be renamed to be more descriptive as its no longer an abstract implementation

File: nav2_smart_planner/include/nav2_smart_planner/smart_planner.hpp [hunk 2]
```diff
@@ -33,16 +33,18 @@
...
 {
 
-class SmartPlanner : public nav2_tasks::ComputePathToPoseTaskServer
+class SmartPlanner : public rclcpp::Node
 {
 public:
   SmartPlanner();
   ~SmartPlanner();
 
   nav2_tasks::TaskStatus execute(
```
Comment (SteveMacenski) [inline nav2_smart_planner/include/nav2_smart_planner/smart_planner.hpp]: If this is no longer an implementation of a task, I think the Execute() method should be renamed to be more descriptive as its no longer an abstract implementation

File: nav2_mission_executor/include/nav2_mission_executor/mission_executor.hpp (hunk 1 above)
Comment (mjeronimo) [reply nav2_mission_executor/include/nav2_mission_executor/mission_executor.hpp]: Sure. I'll update these.

File: nav2_smart_planner/include/nav2_smart_planner/smart_planner.hpp (hunk 2 above)
Comment (mjeronimo) [reply nav2_smart_planner/include/nav2_smart_planner/smart_planner.hpp]: Done.

**Pull request merged**